service-base/src/main/java/org/eclipse/hono/service/quarkus/SamplerProducer.java

```diff
@@ -52,7 +55,12 @@ Sampler sampler(final TracerRuntimeConfig tracerRuntimeConfig) {
         // Dropping all requests spans also makes sure there are no liveness/readiness request spans created.
         // Suppression of these spans via "quarkus.opentelemetry.tracer.suppress-non-application-uris" doesn't work
         // if the non-application-root-path is the same as the overall root path.
-        return new DropHttpRequestSpansSampler(sampler);
+        sampler = new DropHttpRequestSpansSampler(sampler);
+        // drop spans for the given event bus message prefixes
```

Comment:
I wonder if we might want to actually use the non-application root path then. Given that we already decided that the 2.0.0 chart will not be backward compatible with Hono 1.x, I think that we could as well take advantage of the existing features then ...

Reply:
Right, we could do that. (The `DropHttpRequestSpansSampler` would still be needed though for now, as not only the non-application-root-path spans shall get dropped. Issue for the removal of that sampler is #3213.)
Changing the non-application-root-path and the healthcheck endpoints should be done in a separate PR then.

Reply:
> as not only the non-application-root-path spans shall get dropped.

Couldn't we simply remove the TracingHandler in favor of the standard functionality provided by the quarkus-opentelemetry extension?

Reply:
That is basically the goal of #3213 (see also https://github.com/eclipse/hono/pull/3205#discussion_r853736071).

It's a bit more than just removing the `TracingHandler` class. You mean including this in 2.0.0?
I began working on this but then stopped, waiting for the Quarkus update. I can have another look.

Reply:
Well, if we want to use the non-application-root path for probes and metrics, then I think we should do it with 2.0.0. Not sure how this is related to getting rid of the TracingHandler, but if we can kill two birds with one stone, then I think we should do it.

Reply:
In any case, this PR here is mostly about the event bus spans. I've only included a small fix for the `DropHttpRequestSpansSampler` regarding the Quarkus update (and the changed span names there).

Reply:
With regards to the dropping of the healthcheck endpoint spans, the current Hono code is independent from the used non-application-root-path. As a custom sampler is needed in Hono (for using the `SamplingPrioritySampler`), dropping of these spans needs to be done manually anyway (until a solution for https://github.com/quarkusio/quarkus/issues/25525 is implemented). So, mandating usage of the `/q` healthcheck endpoint prefix would only enable us to simplify the span-dropping logic a bit here.

In that sense, I have no strong view about switching to use the `/q` prefix. But I guess, as Quarkus is now used for all Hono components and we don't need Helm-chart backwards-compatibility here, we might as well stick to the Quarkus default paths.